File: src/main/java/com/nickcoblentz/montoya/settings/ShortExtensionSetting.java
package com.nickcoblentz.montoya.settings;

import burp.api.montoya.MontoyaApi;

public class ShortExtensionSetting extends GenericExtensionSetting<Short> {

    public ShortExtensionSetting(MontoyaApi api, String name, String storageKey, Short defaultValue, ExtensionSettingSaveLocation defaultSaveLocation) {
        super(api, name, storageKey, defaultValue, defaultSaveLocation);
    }

    @Override
    public void loadFromProject() {
        setCurrentValue(getApi().persistence().extensionData().getShort(getStorageKey()));
    }

    @Override
    public void loadFromPreferences() {
        setCurrentValue(getApi().persistence().preferences().getShort(getStorageKey()));
    }

    @Override
    public Short parseString(String value) {
        return Short.valueOf(value);
    }

    public void setCurrentValue(String value) {
        if(value!=null)
            setCurrentValue(parseString(value));
    }
}

File: src/main/java/com/nickcoblentz/montoya/settings/ShortExtensionSetting.kt
package com.nickcoblentz.montoya.settings

import burp.api.montoya.MontoyaApi

class ShortExtensionSetting(
    api: MontoyaApi,
    name: String,
    storageKey: String,
    defaultValue: Short,
    defaultSaveLocation: ExtensionSettingSaveLocation
) :
    GenericExtensionSetting<Short>(api, name, storageKey, defaultValue, defaultSaveLocation) {
    override fun loadFromProject() {
        currentValue = api.persistence().extensionData().getShort(storageKey)
    }

    override fun loadFromPreferences() {
        currentValue = api.persistence().preferences().getShort(storageKey)
    }

    override fun parseString(value: String): Short {
        return value.toShort()
    }

    override fun setCurrentValue(value: String?) {
        if (value != null) currentValue = parseString(value)
    }

    override fun saveToProject() {
        api.persistence().extensionData().setShort(storageKey, currentValue)
    }

    override fun saveToPreferences() {
        api.persistence().preferences().setShort(storageKey, currentValue)
    }
}